### ISSUE TITLE
Delay of about 10 seconds before downloading some youtube videos

### ISSUE THREAD

**Issue by triplesixman** [yt-dlp/yt-dlp#4274]
### Checklist

- [X] I'm asking a question and **not** reporting a bug or requesting a feature
- [X] I've looked through the [README](https://github.com/yt-dlp/yt-dlp#readme)
- [X] I've verified that I'm running yt-dlp version **2022.06.29** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar questions **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)

It is very likely that this is related to a change on the youtube side but I was wondering if other people had noticed it.

I notice that when I try to download a video such as this one https://www.youtube.com/watch?v=6Dh-RL__uN4

I get stuck for 10 to 15 seconds at the "[youtube] 6Dh-RL__uN4: Downloading android player API JSON" step, then the download finally starts.

On the other hand with others videos, the delay is less than a second. And of course it is systematic. 

I've noticed this for a few days, maybe less than a week. Has anyone else noticed this?

Example:

Slow https://youtu.be/6Dh-RL__uN4
```
root@local:~/test# time /usr/local/bin/yt-dlp.sh https://youtu.be/6Dh-RL__uN4 --verbose
[debug] Command-line config: ['https://youtu.be/6Dh-RL__uN4', '--verbose']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version 2022.06.29 [9d339c41e] (source)
[debug] Plugins: ['SamplePluginIE', 'SamplePluginPP']
[debug] Git HEAD: 84a251e1f
[debug] Python 3.9.7 (CPython 64bit) - Linux-5.10.0-15-amd64-x86_64-with-glibc2.31 (glibc 2.31)
[debug] Checking exe version: ffmpeg -bsfs
[debug] Checking exe version: ffprobe -bsfs
[debug] exe versions: ffmpeg 4.3.4-0, ffprobe 4.3.4-0, phantomjs 2.1.1
[debug] Optional libraries: sqlite3-2.6.0
[debug] Proxy map: {}
[debug] [youtube] Extracting URL: https://youtu.be/6Dh-RL__uN4
[youtube] 6Dh-RL__uN4: Downloading webpage
[youtube] 6Dh-RL__uN4: Downloading android player API JSON
[debug] Sort order given by extractor: quality, res, fps, hdr:12, source, codec:vp9.2, lang, proto
[debug] Formats sorted by: hasvid, ie_pref, quality, res, fps, hdr:12(7), source, vcodec:vp9.2(10), acodec, lang, proto, filesize, fs_approx, tbr, vbr, abr, asr, vext, aext, hasaud, id
[debug] Default format spec: bestvideo*+bestaudio/best
[info] 6Dh-RL__uN4: Downloading 1 format(s): 248+251
[debug] Invoking http downloader on "https://rr6---sn-a5mlrnes.googlevideo.com/videoplayback?expire=1657026841&ei=ueTDYqWIKZqTkwan45eQBA&ip=XXX&id=o-AN19zSI2RTxCkdYy3-zo9EdICbOG858lICjKOH0ZMleC&itag=248&source=youtube&requiressl=yes&mh=bk&mm=31%2C26&mn=sn-a5mlrnes%2Csn-n4v7snll&ms=au%2Conr&mv=m&mvi=6&pl=24&gcr=us&initcwndbps=60997500&vprv=1&mime=video%2Fwebm&gir=yes&clen=31942785&dur=134.501&lmt=1655017863224615&mt=1657005179&fvip=5&keepalive=yes&fexp=24001373%2C24007246&c=ANDROID&txp=4535434&sparams=expire%2Cei%2Cip%2Cid%2Citag%2Csource%2Crequiressl%2Cgcr%2Cvprv%2Cmime%2Cgir%2Cclen%2Cdur%2Clmt&sig=AOq0QJ8wRQIgPLaVS5oZwIumQWwL6MIawpLtmRRy3QiLpiUq4QVnSoACIQDEMaqkrMoK3gcD_6rDkfTcksUkJgED3yxZrbawGj19WQ%3D%3D&lsparams=mh%2Cmm%2Cmn%2Cms%2Cmv%2Cmvi%2Cpl%2Cinitcwndbps&lsig=AG3C_xAwRQIhAMgxINJledPcS6CjFD60PGDSEiDLzX6og7iNdjdsQHyxAiBSGy_Owxq6hJIxlFhgz-Z8VIoWkVqR8ui82eDLQlYLOw%3D%3D"
[download] Destination: bitch lasagna [6Dh-RL__uN4].f248.webm
[download] 100% of 30.46MiB in 00:00
[debug] Invoking http downloader on "https://rr6---sn-a5mlrnes.googlevideo.com/videoplayback?expire=1657026841&ei=ueTDYqWIKZqTkwan45eQBA&ip=XXX&id=o-AN19zSI2RTxCkdYy3-zo9EdICbOG858lICjKOH0ZMleC&itag=251&source=youtube&requiressl=yes&mh=bk&mm=31%2C26&mn=sn-a5mlrnes%2Csn-n4v7snll&ms=au%2Conr&mv=m&mvi=6&pl=24&gcr=us&initcwndbps=60997500&vprv=1&mime=audio%2Fwebm&gir=yes&clen=2259932&dur=134.521&lmt=1655014274362177&mt=1657005179&fvip=5&keepalive=yes&fexp=24001373%2C24007246&c=ANDROID&txp=4532434&sparams=expire%2Cei%2Cip%2Cid%2Citag%2Csource%2Crequiressl%2Cgcr%2Cvprv%2Cmime%2Cgir%2Cclen%2Cdur%2Clmt&sig=AOq0QJ8wRQIgfAPj9HOSKyB--g5vCMHW-hedvPXivohRc-A3Dk8r0vUCIQC8rRXb48O98jEBvo44KdB5E5k8evujkUwQn0hhlM-w7w%3D%3D&lsparams=mh%2Cmm%2Cmn%2Cms%2Cmv%2Cmvi%2Cpl%2Cinitcwndbps&lsig=AG3C_xAwRQIhAMgxINJledPcS6CjFD60PGDSEiDLzX6og7iNdjdsQHyxAiBSGy_Owxq6hJIxlFhgz-Z8VIoWkVqR8ui82eDLQlYLOw%3D%3D"
[download] Destination: bitch lasagna [6Dh-RL__uN4].f251.webm
[download] 100% of 2.16MiB in 00:00
[Merger] Merging formats into "bitch lasagna [6Dh-RL__uN4].webm"
[debug] ffmpeg command line: ffmpeg -y -loglevel repeat+info -i 'file:bitch lasagna [6Dh-RL__uN4].f248.webm' -i 'file:bitch lasagna [6Dh-RL__uN4].f251.webm' -c copy -map 0:v:0 -map 1:a:0 -movflags +faststart 'file:bitch lasagna [6Dh-RL__uN4].temp.webm'
Deleting original file bitch lasagna [6Dh-RL__uN4].f248.webm (pass -k to keep)
Deleting original file bitch lasagna [6Dh-RL__uN4].f251.webm (pass -k to keep)

real    0m9.659s
user    0m8.424s
sys     0m0.206s
```

Fast https://youtu.be/5QUubSXv6No
```
root@local:~/test# time /usr/local/bin/yt-dlp.sh https://youtu.be/5QUubSXv6No
[youtube] 5QUubSXv6No: Downloading webpage
[youtube] 5QUubSXv6No: Downloading android player API JSON
[info] 5QUubSXv6No: Downloading 1 format(s): 248+251
[download] Destination: POLOZHENIE X DRIVE FOREVER ( ZEDLINE REMIX ) TRAD.FR ft. PEAKY BLINDERS [5QUubSXv6No].f248.webm
[download] 100% of 22.96MiB in 00:00
[download] Destination: POLOZHENIE X DRIVE FOREVER ( ZEDLINE REMIX ) TRAD.FR ft. PEAKY BLINDERS [5QUubSXv6No].f251.webm
[download] 100% of 1.93MiB in 00:00
[Merger] Merging formats into "POLOZHENIE X DRIVE FOREVER ( ZEDLINE REMIX ) TRAD.FR ft. PEAKY BLINDERS [5QUubSXv6No].webm"
Deleting original file POLOZHENIE X DRIVE FOREVER ( ZEDLINE REMIX ) TRAD.FR ft. PEAKY BLINDERS [5QUubSXv6No].f248.webm (pass -k to keep)
Deleting original file POLOZHENIE X DRIVE FOREVER ( ZEDLINE REMIX ) TRAD.FR ft. PEAKY BLINDERS [5QUubSXv6No].f251.webm (pass -k to keep)

real    0m2.463s
user    0m1.014s
sys     0m0.155s
```

By putting --verbose on the video that takes time to download, the 10 second delay occurs between these 2 steps:

```
[debug] Formats sorted by: hasvid, ie_pref, quality, res, fps, hdr:12(7), source, vcodec:vp9.2(10), acodec, lang, proto, filesize, fs_approx, tbr, vbr, abr, asr, vext, aext, hasaud, id
[debug] Default format spec: bestvideo*+bestaudio/best
```

This is quite disabling for the tasks I have to do because it takes me much longer. I had to try it on 3 computers in 3 different locations and it always happens. I would be curious to know if it happens on your side too.

**Comment by coletdjnz** [comment]
verbose logs when you post logs please

I don't think this is network related as there doesn't seem to be any requests during this time :thinking: 

I can repro on master; on my machine the delay is only a few seconds. It occurs after the format sort line is printed (master). CPU usage for the process is maxed out during this time.

cc @pukkandan

**Comment by pukkandan** [comment]
The given example video no longer exists. But if you find another example, try `--extractor-args youtube:skip=translated_subs`

**Comment by coletdjnz** [comment]
> The given example video no longer exists. But if you find another example, try `--extractor-args youtube:skip=translated_subs`

Ah. That flag seems to fix it for me.

Also the video still exists for me (maybe blocked in your region) :thinking:

**Comment by pukkandan** [comment]
> Also the video still exists for me (maybe blocked in your region?) 🤔

That appears to be the case. I can access it over VPN



> > The given example video no longer exists. But if you find another example, try `--extractor-args youtube:skip=translated_subs`
> 
> Ah. That flag seems to fix it for me.

The delay is the time taken by Python to construct the gigantic automatic-captions dictionary. I don't have any idea how it can be sped up - which is part of the reason why this extractor argument was added. You can take a look if you have any ideas. Here's the relevant code: https://github.com/yt-dlp/yt-dlp/blob/ae61d108dd83a951b6e8a27e1fb969682416150d/yt_dlp/extractor/youtube.py#L3602-L3646

Otherwise, our only options are to either (1) ask users to add this option, (2) process subs only if --write-subs etc are given, or (3) Remove translated subs by default and add an extractor-arg to bring it back. (I am not fond of option 3)

**Comment by triplesixman** [comment]
I edited the post with the verbose logs. Indeed, you have put your finger on the problem and it was not related to a request because with the "--extractor-args youtube:skip=translated_subs" parameter it is fast now.

If we don't need the subtitles, I guess you recommend to add this parameter (which must have been added by default in the last update I guess?) ?

I will be using yt-dlp with this setting but if you can update this topic in case you make a change that no longer requires using this default setting, I would greatly appreciate it!

Thanks for your help

**Comment by pukkandan** [comment]
> If we don't need the subtitles, I guess you recommend to add this parameter

Yes

> (which must have been added by default in the last update I guess?) ?

No

**Comment by triplesixman** [comment]
2 last questions about this: According to the yt-dlp logs, the translated subs parameter was added on 2022.04.08, so I can consider that this slowness exists on some videos since all this time even if I noticed it only a few days ago?

I also use yt-dlp to get json data (like title, description...) with the "-j" parameter. With the "--extractor-args youtube:skip=translated_subs" parameter it's also fast again. Is there any problem to use this parameter when I get the json data if I don't want to get the subtitles? I guess not but I would like to have your confirmation!
